tests/test_cli_compile.py

```diff
@@ -2494,9 +2497,275 @@ def test_all_extras_fail_with_extra(fake_dists, runner, make_module, fname, cont
     assert exp in out.stderr
 
 
+def _parse_deps(text: str, keys: Iterable[str]) -> dict[str, str | None]:
+    """Parse ``pip-compile`` output and return mapping from package name to package version
+
+    :param text: output from ``pip-compile``
+    :param keys: names of expected packages. If not in text the version will be set to ``None``.
+    """
+    result: dict[str, str | None] = {k: None for k in keys}
+
+    for line in text.splitlines():
+        prefix = line.split("#")[0].strip()
+        if not prefix:
+            continue
+        try:
+            k, v = prefix.split("==")
+        except ValueError:
+            continue
+        result[k] = v
+
+    return result
+
+
+class Version:
+    def __init__(self, expected: str) -> None:
+        self._pat = expected
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, str):
+            return False
+
+        return fnmatch.fnmatch(other, self._pat)
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}@{id(self)}(_pat={self._pat!r})>"
+
+
+# This can be removed when support for python<3.8 is dropped
+def copytree_dirs_exist_ok(
+    src_top: str | pathlib.Path, dst_top: str | pathlib.Path
```

Comment:
I'm pretty sure that the better type is
```suggestion
    src_top: os.PathLike, dst_top: os.PathLike,
```

Reply:
As far as I can tell `str` is not a `os.PathLike`.
Looking at the type hints for pathlib supports this as it defines and uses `StrPath = Union[str, PathLike[str]]  # stable`.
Do you think the new signature is an improvement?